packages/lodestar/src/db/api/beacon/stateContextCache.ts

```diff
@@ -0,0 +1,54 @@
+import {ByteVector, toHexString, TreeBacked} from "@chainsafe/ssz";
+import {BeaconState} from "@chainsafe/lodestar-types";
+import {EpochContext} from "@chainsafe/lodestar-beacon-state-transition";
+
+export interface IStateContextCacheItem {
+  state: TreeBacked<BeaconState>;
+  epochCtx?: EpochContext;
+}
+
+/**
+ * Simple BeaconState in-memory cache
+ *
+ * Similar API to Repository, but synchronous
+ */
+export class StateContextCache {
```

Comment:
Considering we're always passing passing in both a state and epochCtx, and always needing to get both a state and epochCtx, I don't think its worth having the separate interface with optional epochCtx here, just keep it simple for now.